subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainQueryService.java

```diff
@@ -143,6 +136,9 @@ private JavaToolchain query(JavaToolchainSpec spec, boolean isFallback) {
                 .min(new JavaToolchainComparator());
 
         if (detectedToolchain.isPresent()) {
+            //notify toolchain install service that even thought it didn't now, it might get requests at a later point in time
+            installService.notifyAboutActivity();
```

Comment:
We need to be careful here. But from what I saw, we should be able to figure out the source by looking at the `InstallationLocation.source`

Reply:
Why not just warn when toolchains are being used & auto-provisioning is turned on & there are no repositories defined. That's what I'm aiming for with these changes. Looking at toolchains sources seems messy to me and also seems too much of a restriction.

Reply:
Because if you have a toolchain found from SDKMan, what's the link with auto-provisioning?